.github/workflows/ci-renovate.yml

```diff
@@ -32,20 +32,21 @@ jobs:
           # trigger new workflow runs, but we want to re-run the CI after pushing the updated commit.
           # https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
           token: ${{ secrets.BOT_GITHUB_TOKEN || secrets.GITHUB_TOKEN }}
-      - name: Set workflow variables
-        # Id is needed to access output in a next step.
-        id: vars
-        run: |
-          echo "FILE_NAME=.changelog/${{ github.event.pull_request.number }}.internal.md" >> $GITHUB_OUTPUT
       - name: Create and commit Change Log fragment if it does not exist
+        env:
+          # There's no support for escaping this for use in a shell command.
+          # GitHub's recommendation is to pass it through the environment.
+          # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
+          FILE_NAME: .changelog/${{ github.event.pull_request.number }}.internal.md
+          HEAD_REF: ${{ github.head_ref }}
         run: |
-          if [[ ! -f "${{ steps.vars.outputs.FILE_NAME }}" ]]; then
-            echo "Update dependencies" > ${{ steps.vars.outputs.FILE_NAME }}
+          if [[ ! -f "$FILE_NAME" ]]; then
+            echo "Update dependencies" > "$FILE_NAME"
             # Set git user email and name to match author of the last commit.
             git config --local user.email "$(git log --pretty='%ae' -1)"
             git config --local user.name "$(git log --pretty=format:'%an' -1)"
-            git add ${{ steps.vars.outputs.FILE_NAME }}
+            git add "$FILE_NAME"
             git commit --amend --no-edit
```

Comment:
Not sure who is the owner of git add here, but I would prefer it was another commit. As who ever the owner is, will be the co-author of the commit - and it would be just cleaner, to see which changes were made by this workflow - the downside is useless commit though.

Reply:
this only runs against renovate-bot's commit; and it is currently always failing. I'll leave it :D